src/Compiler/CompilerUtils.cpp

```diff
@@ -373,96 +364,119 @@ std::string getTargetFilename(
   llvm_unreachable("all cases should be handled in switch");
 }
 
+// Apply optimization passes specified by `optPipeline` on the LLVM IR module
+// for the specified target machine and write the resulting bitcode to
+// `outputStream`.
+static int runOptPasses(llvm::Module *llvmModule, llvm::TargetMachine *target,
```

Comment:
Is there a way to run Opt Passes without having to specify each of the passes here? My concern is that if an additional pass was added in the main Opt, we will not see it here unless we also update the code below accordingly. So isn't there a way to call opt using a single function call (where detailed of the options are then maintained by Opt itself)?

Reply:
Unfortunately, there isn't a single function call to replace the binary invocation of opt.  The passes that run in the opt binary are listed in its [500-line long `main()` function](https://github.com/llvm/llvm-project/blob/main/llvm/tools/opt/opt.cpp#L420), and over [multiple files](https://github.com/llvm/llvm-project/tree/main/llvm/tools/opt) in that directory.

So you're right that if a new pass is added to opt, we'd miss it in the function that I added.  Let me quickly check if anyone has expressed interest in refactoring the code for the opt and llc binaries.